modules/preview-react/side-panel/lib/NewSidePanel/hooks/useSidePanelModel.ts

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import {
+  Model,
+  createEventMap,
+  useEventMap,
+  ToModelConfig,
+  useUniqueId,
+} from '@workday/canvas-kit-react/common';
+
+export type SidePanelOrigin = 'start' | 'end';
+export type SidePanelTransitionStates = 'collapsed' | 'expanded' | 'collapsing' | 'expanding';
+
+export type SidePanelState = {
+  collapsedWidth: number | string;
+  expandedWidth: number | string;
+  expanded: SidePanelTransitionStates;
+  labelId: string;
+  origin: SidePanelOrigin;
+  panelId: string;
```

Comment:
I've been using `id` and extrapolating an id from there. Tabs does this. If an application developer wishes to override these ids, they can do so on the subcomponent directly rather than in the model config.

Reply:
I thought about that. I agree. applying an `id` to the element directly feels like the most natural way to override  it. The `panelId` name comes from the original API (it made more sense there).